### PR TITLE
Add devportal rest api changes for API Chat feature

### PR DESCRIPTION
**Pull request by ashera96**
### Purpose

Add devportal REST API changes needed for API Chat feature.

### Approach

We have introduced two resources to handle the API Chat testing capabilities:
- apis/{apiId}/api-chat/prepare
- apis/{apiId}/api-chat/execute

The following configuration is added to the `deployment.toml` to handle feature enablement and configuring.
```toml
[apim.ai]
enable = true
token = "<on-prem key>"
endpoint="http://localhost:9090"

```

### Related PRs

- https://github.com/wso2/product-apim/pull/13422
- https://github.com/wso2/apim-apps/pull/614

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java [hunk 1]
```diff
@@ -501,6 +501,14 @@ public final class APIConstants {
     public static final String DIGEST = "x5t#S256";
     public static final String CNF = "cnf";
 
+    // Constants related to AI features: API chat and Marketplace Assistant
```
Comment (PasanT9) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java:504]: Let' have a subclass for AI related constants.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 2]
```diff
@@ -3308,6 +3308,23 @@ public String getOpenAPIDefinition(String apiId, String organization) throws API
...
+        return APIUtil.invokeAIService(APIConstants.API_CHAT_ENDPOINT,
+                APIConstants.API_CHAT_AUTH_TOKEN, APIConstants.API_CHAT_EXECUTE_RESOURCE, requestPayload,
+                apiChatRequestId);
+    }
+
+    @Override
+    public String invokeApiChatPrepare(String apiId, String apiChatRequestId, String organization)
+            throws APIManagementException {
+        String swaggerDefinition = getOpenAPIDefinition(apiId, organization);
+        String payload = "{\"openapi\": " + swaggerDefinition + "}";
```
Comment (PasanT9) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:3322]: Shall remove the use of magic strings?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 3]
```diff
@@ -10371,4 +10371,76 @@ public static String getScopesAsString(Set<Scope> scopes) {
...
+     * Check whether API Chat feature is enabled
+     *
+     * @return returns true if API Chat feature is enabled, false if disabled.
+     */
+    public static boolean isApiChatEnabled() {
+
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        String isApiChatEnabled = config.getFirstProperty(APIConstants.API_CHAT_ENABLED);
+        if (isApiChatEnabled == null) {
```
Comment (PasanT9) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:10385]: We can handle this using a ternary operator. What do you think?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 4]
```diff
@@ -10371,4 +10371,76 @@ public static String getScopesAsString(Set<Scope> scopes) {
...
+     * Checks whether an auth token is provided for AI features to use. This token is utilized for authentication and
+     * throttling purposes.
+     *
+     * @return returns true if a valid auth token is found, false otherwise.
+     */
+    public static boolean isAuthTokenProvidedForAIFeatures() {
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        String authToken = config.getFirstProperty(APIConstants.API_CHAT_AUTH_TOKEN);
+        if (authToken == null || authToken.equals("")) {
```
Comment (PasanT9) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:10402]: We can use StringUtils.isEmpty(authToken) to handle this.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/v1/dto/ApiChatExecuteRequestApiSpecDTO.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  @Override
+  public int hashCode() {
+    return Objects.hash(serviceUrl, tools);
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder();
+    sb.append("class ApiChatExecuteRequestApiSpecDTO {\n");
+
```
Comment (PasanT9) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/v1/dto/ApiChatExecuteRequestApiSpecDTO.java:89]: Let's remove this new line

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java (hunk 2 above)
Comment (ashera96) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:3322]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12374/files#diff-eea96b2f615001930c3a7cf2ec3802f9dc86ac99cabcd2ae10716b934c76c3a2L3328-R3340

**Pull request merged**